concepts/type-conversion/about.md

```diff
@@ -1,10 +1,272 @@
 # About
 
-JavaScript values can be of many types, when performing operations or comparisons, types need to either be _explicitly converted_ to the same type, or _implicitly coerced_ to the same type.
+In JavaScript, values may be of different types. Changing the type of a variable can be done by explicit _type conversion_.
+Besides that, JavaScript also performs _type coercion_ (implicit type conversion) when the context requires it.
 
-Generally, type coercion is avoided because it is averse to unexpected errors. Instead, opt for practices which allow for explicit type conversion.
+## Type Conversion
 
-An overview of these terms can be found on _MDN Web Docs_:
+JavaScript does not have a construct to cast into a (different) type like many other languages but there are built-in helpers that can be used instead.
+Most notably, the global objects `Boolean`, `Number` and `String` can be used as functions to convert a value to the respective type.
 
-- [Type Conversion](https://developer.mozilla.org/en-US/docs/Glossary/Type_Conversion)
-- [Type Coercion](https://developer.mozilla.org/en-US/docs/Glossary/Type_coercion)
+### Converting to a Boolean (Truthy/Falsy Values)
+
+With `Boolean(value)` you can convert any value to a boolean.
+How does that work?
+
+There is a fixed set of values, so called _falsy_ values, that convert to `false`.
+Most importantly, `false`, `0`, empty string, `null`, `undefined` and `NaN` are falsy.
+The [MDN article on "Falsy"][mdn-falsy] shows the complete list.
+
+For all other values, `Boolean` returns `true`.
+These values are called _truthy_.
+
+```javascript
+Boolean(-1);
+// => true
+
+Boolean(0);
+// => false
+
+Boolean(' ');
+// => true
+
+Boolean('');
+// => false
+```
+
+Note that because of the described rules, the following values are truthy in JavaScript although they might count as falsy in other languages.
+
+```javascript
+Boolean('0');
+// => true
+
+Boolean('false');
+// => true
+
+Boolean([]);
+// => true
+
+Boolean({});
+// => true
+```
+
+### Converting to a Number
+
+`Number(value)` can be used to convert a value to a number.
+Whitespaces at the beginning and the end of a string are ignored and an empty string is converted to `0`.
+If you try to convert a non-primitive value or a string that does not represent a number, **no** error will be thrown.
+Instead, the result is `NaN` ([Not-A-Number][mdn-nan]).
+
+```javascript
+Number('  -12.34  ');
+// => -12.34
+
+Number('1,2');
+// => NaN
+
+Number('1 2');
+// => NaN
+
+Number('');
+// => 0
+
+Number('10e3');
+// => 10000
+
+Number({ key: '123' });
+// => NaN
+```
+
+Below you can see what `Number` returns for other primitive values.
+
+```javascript
+Number(true);
+// => 1
+
+Number(false);
+// => 0
+
+Number(null);
+// => 0
+
+Number(undefined);
+// => NaN
```

Comment:
So `Number(undefined)` is `NaN`, but `Number()` is `0`? 🤯

Reply:
you are right but I didn't know this either :exploding_head:

Reply:
I added a sentence about this behavior. I couldn't find an explanation though.
@SleeplessByte Do you know what's going on there?

Reply:
I do. It's specced

```
Number ( [ value ] )

Returns a Number value (not a Number object) computed by ToNumber(value) if value was supplied, else returns +0.
```

So this has a special case for _no_ argument, which is `+0`.

Then we check `ToNumber`:

Argument Type | Result
-- | --
Undefined | NaN
Null | +0
Boolean | The result is 1 if the argument is true. The result is +0 if the argument is false.
Number | The result equals the input argument (no conversion).
String | See grammar and note below.
Object | Apply the following steps:Let primValue be ToPrimitive(input argument, hint Number).Return ToNumber(primValue).

and _thus_ explicitly passing `undefined` is a different result (`NaN`) than passing _nothing_.

Reply:
I got that part but I was wondering why it was specced that way.

Reply:
Probably legacy / `new String`, `new Array`, `new Number` should be "the same" as in create an "empty" value of that type (`''`, `[]`, and `+0`). It would be weird if that created a `NaN`. However passing in `undefined` makes it subject to coercion. Something like that.